annotationPipeline/src/main/java/org/cbioportal/annotation/pipeline/MutationRecordReader.java

```diff
@@ -131,61 +137,63 @@ public void handleLine(String line) {
+            String serverErrorMessage = "";
             AnnotatedRecord annotatedRecord = new AnnotatedRecord(record);
             try {
                 annotatedRecord = annotator.annotateRecord(record, replace, isoformOverride, true);
             }
             catch (HttpServerErrorException ex) {
-                String reasonFailed = "Failed to annotate variant due to internal server error";
-                LOG.warn(reasonFailed + ": " + variantDetails);
-                updateErrorMessages(record, record.getVARIANT_CLASSIFICATION(), annotator.getUrlForRecord(record, isoformOverride), reasonFailed);
+                serverErrorMessage = "Failed to annotate variant due to internal server error";
             }
             catch (HttpClientErrorException ex) {
-                String reasonFailed = "Failed to annotate variant due to client error";
-                LOG.warn(reasonFailed + ": " + variantDetails);
-                updateErrorMessages(record, record.getVARIANT_CLASSIFICATION(), annotator.getUrlForRecord(record, isoformOverride), reasonFailed);                
+                serverErrorMessage = "Failed to annotate variant due to client error";
             }
             catch (HttpMessageNotReadableException ex) {
-                String reasonFailed = "Failed to annotate variant due to message not readable error";
-                LOG.warn(reasonFailed + ": " + variantDetails);
-                updateErrorMessages(record, record.getVARIANT_CLASSIFICATION(), annotator.getUrlForRecord(record, isoformOverride), reasonFailed);                 
+                serverErrorMessage = "Failed to annotate variant due to message not readable error";
             }
             catch (GenomeNexusAnnotationFailureException ex) {
-                String reasonFailed = "Failed to annotate variant due to Genome Nexus : " + ex.getMessage();
-                LOG.warn(reasonFailed + ": " + variantDetails);
-                updateErrorMessages(record, record.getVARIANT_CLASSIFICATION(), annotator.getUrlForRecord(record, isoformOverride), reasonFailed);                 
+                serverErrorMessage = "Failed to annotate variant due to Genome Nexus : " + ex.getMessage();
+            }
+            annotatedRecords.add(annotatedRecord);
+            header.addAll(annotatedRecord.getHeaderWithAdditionalFields());
+            
+            // log server failure message if applicable
+            if (!serverErrorMessage.isEmpty()) {
+                LOG.warn(serverErrorMessage);
+                if (errorReportLocation != null) updateErrorMessages(record, record.getVARIANT_CLASSIFICATION(), annotator.getUrlForRecord(record, isoformOverride), serverErrorMessage);
+                continue;
+            }
+            String annotationErrorMessage = "";
+            if (MafUtil.variantContainsSnpAndIndel(record.getREFERENCE_ALLELE(), record.getTUMOR_SEQ_ALLELE1(), record.getTUMOR_SEQ_ALLELE2())) {
```

Comment:
As the cbioportal core function is currently implemented, I'm not sure if the use here covers all desired cases. The core function looks for tumorSeqAllele1.equals("-") ... but theoretically a case like this would not be detected for error logging:
refAllele: "AAACCCGGG" (codes for "KPG")
tumorSeqAllele1 : "AAA-GGG" (codes for "KG")
tumorSeqAllele2 : "AAATCCGGG" (codes for "KSG")
Maybe this is very unlikely to appear in the input? Also I'm not sure which mutation should be selected and which should be ignored (The SNP, or the in-frame deletion)
Or maybe this will never appear in the input .. because any flanking nucleotides common to both refAllele and tumorSeqAllele1/2 will be stripped off in the reported variation ... so for the above case we would only ever get:
"CCC"
"-"
"TCC"
?